### PR TITLE
Added -T option for packing from a file listing.

### PR DESCRIPTION
**Pull request by zvonler**
Was looking for a bit more flexibility when adding files from a directory that might contain unwanted files. Added a small test framework too, it may need some adjustments to be platform independent, as of now I've only tested on MacOS.

### CODE REVIEW

**Review by earlephilhower — CHANGES_REQUESTED** [review]
First of all, thanks very much for adding tests!  One minor thing Win32 related here, I think.

File: main.cpp [hunk 1]
```diff
@@ -231,6 +232,34 @@ int addFile(char* name, const char* path) {
...
+        return 1;
+    }
+
+    char* srcpath = NULL;
+    size_t len = 0;
+    bool error = false;
+    ssize_t linelen;
+    while ((linelen = getline(&srcpath, &len, listing)) != -1) {
+        if (!linelen) continue;
+        if (srcpath[linelen - 1] == '\n') {
```
[the remark this answers is not in the document]
Comment (zvonler) [reply main.cpp:248]: Sure, didn't know if supporting DOS files without complaint was desired - I'll update.

**Comment by earlephilhower** [comment]
Sorry, I just got back from a business trip last night.  I'll look at this in the next day or two.  Thanks for adding Windows support!

**Comment by zvonler** [comment]
> Sorry, I just got back from a business trip last night. I'll look at this in the next day or two. Thanks for adding Windows support!

Right on, no problem.

**Review by earlephilhower — APPROVED** [review]
Thanks!

**Comment by Jason2866** [comment]
@earlephilhower Fails to compile with Windows. Used the docker build env
See GH Actions run https://github.com/Jason2866/mklittlefs/actions/runs/6284288146/job/17065511128

**Pull request merged**